### ISSUE TITLE
UDDSketch assertion at crates/udd-sketch/src/lib.rs:290:9

### ISSUE THREAD

**Issue by tyhoff** [timescale/timescaledb-toolkit#396]
**Relevant system information:**
 - OS: Aiven's TimescaleDB provider
 - PostgreSQL version, 13.6
 - TimescaleDB Toolkit version: 1.5.1
 - Installation method: Aiven

**Describe the bug**
I hit an assert when I don't expect to when creating a UDDSketch.

**To Reproduce**
Apologies for the seemingly random test case, but here it goes.

```
CREATE TABLE "public"."timescale_bug" ("n" float8 NOT NULL);
```

Load the attached CSV file into the table. 

Run

```
SELECT
	uddsketch(100, 0.01, CAST(n AS FLOAT))
FROM
	timescale_bug
```

**Expected behavior**
Obvious

**Actual behavior**
```
Query 1 ERROR: ERROR:  assertion failed: (self.gamma.powf(1.0 / f64::powi(2.0, self.compactions as i32)) -\n     other.gamma.powf(1.0 / f64::powi(2.0, other.compactions as i32))).abs() <\n    f64::EPSILON
CONTEXT:  crates/udd-sketch/src/lib.rs:290:9
```

Fudging with the bucket count or error will get it to work, but it's random when the assert is tripped.

[timescale_bug.csv.zip](https://github.com/timescale/timescaledb-toolkit/files/8524661/timescale_bug.csv.zip)

**Comment by bvanelli** [comment]
What is the status on this one? Would an upgrade to > 1.60 solve this issue? I'm having the same bug.

**Comment by rtwalker** [comment]
We haven't included a fix for this bug yet. From what I can tell, it appears likely that at least one of the assertions we are making when combining the UDDSketches from different parallel workers is too sensitive to floating point arithmetic.

I will note that in testing this out, I haven't been able to trigger this bug after running 
`SET max_parallel_workers_per_gather = 0;`

I'll be sure to update this issue once we've released a bug fix.

**Comment by tyhoff** [comment]
@rtwalker Thanks for digging into this one! Excited to try it out in the next release.

**Comment by jerryxwu** [comment]
@tyhoff If you haven't noticed, we've just included the fix in the latest 1.8.0 release. We'd love to hear whether it's fixed for you or not.

**Comment by tyhoff** [comment]
> @tyhoff If you haven't noticed, we've just included the fix in the latest 1.8.0 release. We'd love to hear whether it's fixed for you or not.

@jerryxwu I noticed! I'm still waiting for it to be published to Aiven (Timescale Managed) sadly.
